### DISCUSSION TITLE
Cannot install with "frozen-lockfile" on build

### DISCUSSION THREAD

**Discussion by elarion83** [vercel/community#2584] [Help] [answered]
### Summary

Hi, I'm trying to build my React App trought Vercel and I have this error on the Build step : 

 ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up-to-date with package.json

Can anyone help me ?

Thanks a lot :)

**Comment by amyegan** [comment, marked as answer]
Hi @elarion83. You can try deleting the old lockfile and running `pnpm install` to generate a new one. Or try using one of the available [`pnpm install` options](https://pnpm.io/cli/install) to update or repair the lockfile.

**Reply by ngtrongnguyen** [reply]
thanks a lot

**Reply by snolan-ethika** [reply]
what if you're NOT using pnpm in your git repo, but this is due to ONLY vercel using it?

**Reply by ierdss** [reply]
This works for me on Next.js. Thanks a lot!

**Reply by zaki031** [reply]
> what if you're NOT using pnpm in your git repo, but this is due to ONLY vercel using it?

just delete the file and push again

**Reply by snolan-ethika** [reply]
> > what if you're NOT using pnpm in your git repo, but this is due to ONLY vercel using it?
> 
> just delete the file and push again

delete what file? how do you delete a file you don't have in your repo?

**Comment by AuroraLantean** [comment]
Delete pnpm-lock.yaml file and push it into Github for Vercel to redeploy

**Reply by ChiaYuSu** [reply]
this work for me

**Reply by DangNghia17** [reply]
it work for me, thanks so much

**Comment by yasdev08** [comment]
none of these work for me :/

**Reply by Makoehle** [reply]
Make sure to use the correct version. `pnpm -v` has to match the github action.

```
    - uses: pnpm/action-setup@v3
      with:
        version: 9.0.5
```

**Reply by yotlaire** [reply]
That works for me> thank you!

**Reply by hoangducsteven** [reply]
would you go into more details? thanks

**Reply by andrehadianto** [reply]
Look at your GitHub workflow setup (either on your project or your repo) and make sure that the `/action-setup@vX` is used with the same version as your local `pnpm` version.

You can check your local `pnpm` version by running `pnpm -v`

**Comment by snolan-ethika** [comment]
what if you're not using pnpm in your git repo? this seems like an issue on vercel vs. development environment... how do you delete the pnpm lock file from vercel?

**Comment by goomoonarch** [comment]
None of this worked for me, 😎 fortunately I resolved it in the initial configuration before deploying to Vercel, I set the installation code to "npm install", usually this parameter comes by default "pnpm install" but there is an option to adjust it.

[a comment by an automated account is not shown]
